unittests/rpc/RPCHelpersTest.cpp

```diff
@@ -0,0 +1,420 @@
+constexpr static auto ACCOUNT2 = "rLEsXccBGNR3UPuPu2hUXPjziKC3qKSBun";
+constexpr static auto INDEX1 =
+    "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC321";
+constexpr static auto INDEX2 =
+    "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC322";
+constexpr static auto TXNID =
+    "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC321";
+
+class RPCHelpersTest : public MockBackendTest
+{
+    void
+    SetUp() override
+    {
+        MockBackendTest::SetUp();
+    }
+    void
+    TearDown() override
+    {
+        MockBackendTest::TearDown();
+    }
+
+protected:
+    boost::asio::io_context ctx;
+};
+
+TEST_F(RPCHelpersTest, TraverseOwnedNodesNotAccount)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    // fetch account object return emtpy
+    ON_CALL(*rawBackendPtr, doFetchLedgerObject)
+        .WillByDefault(Return(std::optional<Blob>{}));
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObject).Times(1);
+
+    boost::asio::spawn(ctx, [this](boost::asio::yield_context yield) {
+        auto account = GetAccountIDWithString(ACCOUNT);
+        auto ret = traverseOwnedNodes(
+            *mockBackendPtr, account, 9, 10, "", yield, [](auto sle) {
+
+            });
+        auto status = std::get_if<Status>(&ret);
+        EXPECT_TRUE(status != nullptr);
+        EXPECT_EQ(*status, RippledError::rpcACT_NOT_FOUND);
+    });
+    ctx.run();
+}
+
+TEST_F(RPCHelpersTest, TraverseOwnedNodesMarkerInvalid)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    // fetch account object return something
+    auto fake = Blob{'f', 'a', 'k', 'e'};
+    ON_CALL(*rawBackendPtr, doFetchLedgerObject).WillByDefault(Return(fake));
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObject).Times(1);
+
+    boost::asio::spawn(ctx, [this](boost::asio::yield_context yield) {
+        auto account = GetAccountIDWithString(ACCOUNT);
+        auto ret = traverseOwnedNodes(
+            *mockBackendPtr, account, 9, 10, "nothex,10", yield, [](auto sle) {
+
+            });
+        auto status = std::get_if<Status>(&ret);
+        EXPECT_TRUE(status != nullptr);
+        EXPECT_EQ(*status, ripple::rpcINVALID_PARAMS);
+        EXPECT_EQ(status->message, "Malformed cursor");
+    });
+    ctx.run();
+}
+
+TEST_F(RPCHelpersTest, TraverseOwnedNodesMarkerInvalid2)
+{
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    // fetch account object return something
+    auto fake = Blob{'f', 'a', 'k', 'e'};
+    ON_CALL(*rawBackendPtr, doFetchLedgerObject).WillByDefault(Return(fake));
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObject).Times(1);
+
+    boost::asio::spawn(ctx, [this](boost::asio::yield_context yield) {
+        auto account = GetAccountIDWithString(ACCOUNT);
+        auto ret = traverseOwnedNodes(
+            *mockBackendPtr, account, 9, 10, "nothex,abc", yield, [](auto sle) {
```

Comment:
`auto sle` will give us unused variable error later on. let's remove `sle` here and in other instances to prevent it erroring out in the future?